### DISCUSSION TITLE
"Unable to Arm / Sensors Calibrating" needs to be more verbose after a time

### DISCUSSION THREAD

**Discussion by FireNWater** [iNavFlight/inav#9711] [General]
## Current Behavior
When the FC cannot arm due to a sensor problem, the OSD message "Unable to Arm / Sensors Calibrating" shows on the OSD.

## Desired Behavior
Since the flight vehicle could have many sensors, I would like the message to show which sensor is currently keeping the system from arming.

## Suggested Solution
After a period of time, perhaps 30 to 60 seconds, add text to the OSD error message to identify which sensor is preventing the system from arming .

## Who does this impact? Who is this for?
All users.

## Additional context
This feature would greatly enhance troubleshooting, especially at the field.

**Comment by Pairan** [comment]
Had the same in the field lately ... "Unable to Arm / Sensors Calibrating" and then sometime later "RIGHT PITCH NOT CENTERED" ...

I had connected a 4s instead of a 3s battery ... changing that solved my problem somehow,

**Comment by Tgoraj** [comment]
Ok, so plugging a 3s battery in solved it but my plane only uses 4s batteries so… what do I do?why is there no solution to this issue?

**Comment by MrD-RC** [comment]
Hold the aircraft still. Its calibrating the onboard sensors. It can’t calibrate the gyro if it’s moving. 

You’ll probably find the battery is a red herring. But, if not, check the configuration page and what your settings are for the battery.

**Reply by Tgoraj** [reply]
Yes, the battery thing seemed suspect to me too, but I tried to “tighten up” my voltage and current settings anyway with no luck. I’m pretty sure the plane isn’t moving at all, but I even tried being extra careful plugging the battery in and it’s not moving at all. I wonder if the FC is flaky? I can usually find the answers to all of my quirky iNAV issues but this one has practically no answers out there. I’m using the Matek H742 v3 FC

**Reply by Tgoraj** [reply]
The issue ended up being the airspeed sensor. I’m not sure why it is an issue now, I have been running this same setup for a long time. It was an old analog sensor that I had hard-wired. I am not sure why I didn’t think to remove it before, but there you go. If we all could think of everything, there would be no need for forums like this! Switched to a new digital sensor and all good now.

**Comment by sensei-hacker** [comment]
FYI for anyone having a similar problem in the future, the CLI command "status" should show the status of each sensor.
